recipe/gen_patch_json.py

```diff
@@ -401,6 +401,10 @@ def _gen_new_index(repodata, subdir):
             #         depends.append("blas 1.* openblas")
             #         instructions["packages"][fn]["depends"] = depends
 
+        if any(dep.startswith("snappy >=1.1.7") for dep in deps):
```

Comment:
```suggestion
        if any(dep=='snappy >=1.1.7,<1.1.8.0a0' for dep in deps):
```
Otherwise, you might get a keyerror below.